jcheck/src/main/java/org/openjdk/skara/jcheck/ReviewersConfiguration.java

```diff
@@ -76,6 +99,10 @@ public boolean shouldCheckBackports() {
         return shouldCheckBackports;
     }
 
+    public String getReviewRequirements() {
+        return reviewRequirements;
```

Comment:
I would probably have put the logic for generating this string right here in the getter. Are we ever calling this more than once? Could also generate it lazily.

I realize this gets a lot of testing through the ReviewersTest above, but it would make sense to have cheap and quick tests of just this functionality here in the jcheck package.

Reply:
> I would probably have put the logic for generating this string right here in the getter. 

Fixed. Please note the field `reviewRequirements` is not `final` now.

> I realize this gets a lot of testing through the ReviewersTest above, but it would make sense to have cheap and quick tests of just this functionality here in the jcheck package.

Added tests.

Reply:
That's fine, thanks!

I now realize that you put the space and parentheses in the string here. I think this method should just return the requirements and let the visitor add the surrounding formatting.

Reply:
Fixed.

Reply:
@lgxbslgx I missed reviewing this when it was open, and am just going through the backlog of changes that has happened now.

However, I think this was incorrect. The code here makes the text appears as: 
```
Change must be properly reviewed (1 review required, with at least 1 reviewer)
```

when it should have been:
```
Change must be properly reviewed (1 review required, with at least 1 Reviewer)
```

(Note the capitalization of Reviewer).

Traditionally, the (not very clear, I admit) distinction has been made in OpenJDK between "reviewer" (anyone who offers a review) and a "Reviewer" (a member with the official status as Reviewer for a project). This text is about the latter, so changing the "r" to a "R" in the message made a significant difference.

Can you please fix this in a follow-up PR?

Reply:
Do the `contributor`, `author`, `committer`, `lead` also need to be changed to `Contributor`, `Author`, `Committer`, `Lead`?

Reply:
Filed https://bugs.openjdk.java.net/browse/SKARA-1437 to follow up.

Reply:
Yes, they should. See https://openjdk.java.net/bylaws#_7 for official roles.

However, to specify any requirement other than Reviewer seems to me to be mostly an academic exercise. In practice, no-one will ever set any limitation other than the project role as Reviewer. (I'm not even sure why that code is there in the first place. Just some completionist thinking, I believe.)

Reply:
Actually, it's quite common to have as a requirement 2 reviewers, at least one of which must be a Reviewer and the other an Author (i.e., any role in the Project). Both `jdk` and `jfx` are configured this way. Here is the current Progress item that Skara adds in response to the command `/reviewers 2`:

```
- [ ] Change must be properly reviewed (2 reviews required, with at least 1 reviewer, 1 author)
```

And here is the comment:

```
The total number of required reviews for this PR (including the jcheck configuration and the last /reviewers command) is now set to 2 (with 1 of role reviewers, 1 of role authors).
```

Both "Reviewer" and "Author" should be capitalized in both messages. Separately, I see that the pluralization was fixed in the Progress item, but not the comment.

Reply:
There are also projects that do not have any members in the Reviewer role, so they configure their repository to only require Committers. The only role I doubt we will ever see required in reality is Lead, but that's ok.

Reply:
What about changing the message `(with 1 of role reviewers, 1 of role authors)` to `(with at least 1 Reviewer, 1 Author)` and obeying the same pluralization rule. The message `(with 1 of role reviewers, 1 of role authors)` may be fixed as `(with 1 of role Reviewer, 1 of role Author)`. But if there is a message `(with 2 of role reviewers)`, we don't know whether it should fixed as `(with 2 of role Reviewer)` or `(with 2 of role Reviewers)` (pluralization).

Reply:
I submit a PR https://github.com/openjdk/skara/pull/1319. We can continue the discussion there.